src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -1897,64 +1956,79 @@ private long EnforceConfigurationFor(DocumentsOperationContext context, string i
             }
         }
 
-        private class EnforceRevisionConfigurationCommand : TransactionOperationsMerger.MergedTransactionCommand
+        internal long AdoptOrphanedFor(DocumentsOperationContext context, string id)
         {
-            private readonly RevisionsStorage _revisionsStorage;
-            private readonly List<string> _ids;
-            private readonly bool _includeForceCreatedRevisionsOnDeleteInCaseOfNoConfiguration;
-            private readonly EnforceConfigurationResult _result;
-            private readonly OperationCancelToken _token;
+            using (DocumentIdWorker.GetSliceFromId(context, id, out var lowerId))
+            using (GetKeyPrefix(context, lowerId, out var lowerIdPrefix))
+            {
+                var collectionName = GetCollectionFor(context, lowerIdPrefix);
+                if (collectionName == null)
+                {
+                    if (_logger.IsInfoEnabled)
+                        _logger.Info($"Tried to delete revisions for '{id}' but no revisions found.");
+                    return 0;
+                }
 
-            public bool MoreWork;
 
-            public EnforceRevisionConfigurationCommand(
-                RevisionsStorage revisionsStorage,
-                List<string> ids,
-                EnforceConfigurationResult result,
-                bool includeForceCreated,
-                OperationCancelToken token)
-            {
-                _revisionsStorage = revisionsStorage;
-                _ids = ids;
-                _result = result;
-                _token = token;
-                _includeForceCreatedRevisionsOnDeleteInCaseOfNoConfiguration = includeForceCreated;
+                if (ShouldAdoptRevision(context, lowerId, lowerIdPrefix, collectionName, out var table, out var lastRevision))
+                {
+                    var lastModifiedTicks = _database.Time.GetUtcNow().Ticks;
+                    return CreateDeletedRevision(context, table, id, collectionName, lastModifiedTicks, lastRevision.Flags) - 1;
```

Comment:
this returns the number or revisions for a given document, but eventually we want to count how many _docs_ were adopted..
so simply do `return 1`

Reply:
Done